netann/interface.go

```diff
@@ -0,0 +1,23 @@
+package netann
+
+import (
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/channeldb"
+)
+
+// DB abstracts the required database functionality needed by the
+// ChanStatusManager.
+type DB interface {
+	// FetchAllOpenChannels returns a slice of all open channels known to
+	// the daemon. This may include private or pending channels.
+	FetchAllOpenChannels() ([]*channeldb.OpenChannel, error)
```

Comment:
@wpaulino because then we need to wrap the channeldb call in some other struct/closure that implements this. unless you're also suggesting add `FetchPublicOpenChannels` to the db as well